pkg/fqdn/cache.go

```diff
@@ -266,8 +410,10 @@ func (c *DNSCache) updateWithEntryIPs(entries ipEntries, entry *cacheEntry) {
 		if old == nil || !exists || old.isExpiredBy(entry.ExpirationTime) {
 			entries[ipStr] = entry
 			c.upsertReverse(ipStr, entry)
+			c.addNameToCleanup(entry)
 		}
 	}
+
 }
 
 // removeExpired removes expired (or nil) cacheEntry pointers from entries, an
```

Comment:
Since we know the entry we are expiring, we can also remove it from cleanup. This way if someone forgets to run GC we won't leak memory.